### DISCUSSION TITLE
Looking for advice on implementing array of UUIDs or having multiple UUIDs in one column

### DISCUSSION THREAD

**Discussion by brysonreese** [sqlalchemy/sqlalchemy#7754] [Usage Questions]
Hello! I was hoping to get a little bit of advice on a sane way to implement a column that can store multiple UUIDs, which can have more UUIDs added or removed as necessary. I am using PostgreSQL and Flask to write a backend for an application I am working on.

I am autogenerating UUIDs in a table that stores user info called 'users'. My application has groups of users that will be sharing the same information in one row on another table called 'groups', which also have their own autogenerated UUIDs. In the 'users' table, each individual user has a column for whatever the group UUID they belong to is, and that will be a foreign key referencing the group. I am attempting to then be able to store all the users that belong to a group in a column for that specific group, hence the need to store multiple UUIDs in one column.

Here is a little graphical representation of what I mean.
<img width="604" alt="Screen Shot 2022-02-22 at 1 59 53 PM" src="https://user-images.githubusercontent.com/44021258/155226785-1abec25b-5b38-4b8d-b0cb-6729c9b1aefa.png">

I read conflicting reports on whether SQLAlchemy can support an array of UUIDs effectively, to where I would be able to add/remove UUIDs from that members column of the group table.

If anyone has any advice on a method to implement this or knows if this is possible, I would greatly appreciate it! Thank you.

**Comment by zzzeek** [comment]
ARRAY of UUID is available by combining the [ARRAY](https://docs.sqlalchemy.org/en/14/core/type_basics.html?highlight=array#sqlalchemy.types.ARRAY) datatype with the PostgreSQL [UUID](https://docs.sqlalchemy.org/en/14/dialects/postgresql.html?highlight=uuid#sqlalchemy.dialects.postgresql.UUID) datatype. like:

```py

    from sqlalchemy import ARRAY
    from sqlalchemy.dialects.postgresql import UUID

    class Group(Base):
        __tablename__ = 'groups'
        gid = Column(Integer, primary_key=True)
        members = Column(ARRAY(UUID(as_uuid=True)))

```

and that's pretty much it.  Relationally I'd prefer to use a separate table "group_members" with these UUIDs one per row, but PG ARRAY is certainly extremely popular and I'm not the one designing your schema, so it should work out fine.
